tests/std/include/range_algorithm_support.hpp

```diff
@@ -590,7 +591,7 @@ template <class Category, class Element, ::test::CanDifference Diff, ::test::Can
 struct std::iterator_traits<::test::iterator<Category, Element, Diff, Eq, Proxy, Wrapped>> {
     using iterator_concept  = Category;
     using iterator_category = conditional_t<derived_from<Category, forward_iterator_tag>, //
-        conditional_t<static_cast<bool>(Proxy), input_iterator_tag, Category>, //
+        conditional_t<Proxy != ::test::ProxyRef::no, input_iterator_tag, Category>, //

```

Comment:
Would it be simpler to flip the negation? Ok to keep as is.

Reply:
`test::ProxyRef` was a "fancy `bool`" but becomes a tristate `yes`/`no`/`prvalue` in this PR^1. Flipping this condition produces `Proxy == ::test::ProxyRef::yes || Proxy == ::test::ProxyRef::prvalue`, which I find less readable.

1. Yes, this is a horrible hack, but it adds the functionality we need to exercise `join_view` without requiring any changes to other tests that use `test::range`. One of my followup tasks is to replace it with an improved `enum class RefStyle { lvalue, xvalue, prvalue, proxy }` and clean up my mess.

Reply:
What about literally flipping it like this (which is what I was imagining):

`conditional_t<Proxy == ::test::ProxyRef::no, Category, input_iterator_tag>`

Reply:
🤦 Oh, or that.

```suggestion
        conditional_t<Proxy == ::test::ProxyRef::no, Category, input_iterator_tag>, //
```